scripts/docs-deploy/clone-docs-repo.ts

```diff
@@ -0,0 +1,62 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import {$} from 'zx';
+import {projectDir} from './utils';
+
+/** Git repository HTTP url pointing to the docs repository. */
+export const docsRepoUrl = 'https://github.com/angular/material.angular.io.git';
+
+/**
+ * Clones the docs repository for the given major into a
+ * temporary directory.
+ *
+ * @returns An absolute path to the temporary directory.
+ */
+export async function cloneDocsRepositoryForMajor(major: number): Promise<string> {
+  const repoTmpDir = path.join(projectDir, 'tmp/docs-repo');
```

Comment:
Should this be done in a "real" temp director, `os.tmpdir()`?

Reply:
I'm always a little hesitant about this since I don't know how well it works on different platforms. We've been using `<project>/tmp` for the deploy scripts as well (it's great having this stuff local associated with the comp repo)